oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/UserAgentImpl.java

```diff
@@ -110,6 +108,18 @@ AuthorizationResponse encode(final String methodName, final String... parameters
         catch (final InterruptedException e) {
             throw new AuthorizationException("interrupted_exception", e.getMessage(), null, e);
         }
+        catch (final AuthorizationException e) {
+            if ("parsing_error".equals(e.getCode()) && coordinator != null) {

```

Comment:
It seems to me we could avoid filtering the `AuthorizationException` here by having provided the `errorContents` as a second parameter to `AuthorizationResponse#fromString` to initialize its `errorDescription` variable with.